File: problems/dna-interactive/sol.py
import sys
import random

random.seed(0)

f = sys.stdin

E = f.readline().rstrip()
n = len(E)

iters = 100
qlen  = 50

for i in range(100):
    s = random.randint(0, n - qlen)
    print(E[s : s + qlen])
    sys.stdout.flush()
    ans = f.readline().rstrip()
    if ans == "Yes":
        print("Earthovirus")
        sys.stdout.flush()
        exit(0)

print("Marsovirus")
sys.stdout.flush()

File: problems/dna-interactive/sol.cpp
#include <iostream>
#include <cstdlib>
#include <string>
using namespace std;

int main() {
    string E;
    cin >> E;
    int n = E.size();

    int iters = 100;
    int qlen = 50;

    for (int i = 0; i < iters; ++i) {
        int s = rand() % (n - qlen + 1);
        cout << E.substr(s, qlen) << endl << flush;
        string ans;
        cin >> ans;
        if (ans == "Yes") {
            cout << "Earthovirus" << endl << flush;
            return 0;
        }
    }
    cout << "Marsovirus" << endl << flush;
    return 0;
}
